channeld/channeld.c

```diff
@@ -2656,6 +2656,14 @@ static void init_channel(struct peer *peer)
 	 * create error packet if necessary. */
 	init_shared_secrets(peer->channel, htlcs, hstates);
 
+	/* We don't need these any more, so free them. */
+	tal_free(htlcs);
+	tal_free(hstates);
+	tal_free(fulfilled);
+	tal_free(fulfilled_sides);
+	tal_free(failed);
+	tal_free(failed_sides);
```

Comment:
Any chance we can group them into a common parent?

Reply:
Almost.  They're allocated by `fromwire_channel_init`, off `peer`.  But it seems we want to discard most of them so we should allocate off tmpctx and explicitly steal the ones we actually want to keep...